locksmith/src/operations/checkoutConfigOperations.ts

```diff
@@ -8,35 +13,101 @@ interface SaveCheckoutConfigArgs {
   config: any // TODO: TYPE
 }
 
-// Creates or updates a checkout config
+/**
+ * Extracts lock addresses and network information from the config
+ * @param config - The checkout configuration object
+ * @returns An array of objects containing lock address and network
+ */
+const extractLockInfo = (
+  config: any
+): { address: string; network: number }[] => {
+  return Object.entries(config.locks).map(
+    ([address, lockConfig]: [string, any]) => ({
+      address,
+      network: lockConfig.network,
+    })
+  )
+}
+
+/**
+ * Checks if a user is a manager for a given lock
+ * @param lockAddress - The address of the lock
+ * @param userAddress - The address of the user
+ * @param network - The network ID
+ * @returns A boolean indicating whether the user is a lock manager
+ */
+const isLockManager = async (
+  lockAddress: string,
+  userAddress: string,
+  network: number
+): Promise<boolean> => {
+  try {
+    return await web3Service.isLockManager(lockAddress, userAddress, network)
+  } catch (error) {
+    console.error(`Error checking lock manager status: ${error}`)
+    return false
+  }
+}
+
+/**
+ * Determines if a user is authorized to manage a checkout config
+ * @param userAddress - The address of the user
+ * @param checkoutConfig - The checkout configuration object
+ * @returns A boolean indicating whether the user is authorized
+ */
+const isUserAuthorized = async (
+  userAddress: string,
+  checkoutConfig: CheckoutConfig
+): Promise<boolean> => {
+  const lockInfo = extractLockInfo(checkoutConfig.config)
+  for (const { address, network } of lockInfo) {
+    if (await isLockManager(address, userAddress, network)) {
+      return true
+    }
+  }
+  // Fallback: check if user is the creator
+  return checkoutConfig.createdBy === userAddress
```

Comment:
Do we want to do that first since will be faster?

Reply:
I guess so!